frontend/integration-tests/views/metalkube/dashboards.view.ts

```diff
@@ -0,0 +1,168 @@
+ * items of the resource type (for example: "6 Nodes")
+ * @example
+ * await dashboardView.inventoryItemLabel(0).getText()
+ */
+export function inventoryItemLabel(rowNumber) {
+  return element(by.xpath(`(//div[@class="kubevirt-inventory__row-title"])[${rowNumber + 1}]`));
+}
+
+/**
+ * The element in the inventory card, next to the up icon (green check icon) in the nth row.
+ * This element doesn't always exist, so check if it's present first, or use the getTextIfPresent
+ * utility function.
+ * @example
+ * await dashboardView.inventoryUpCounter(0).getText()
+ */
+export function inventoryUpCounter(rowNumber) {
+  // Explaining the xpath:
+  // There is an array of 6 kubevirt-inventory__row-status items (one for each line in the inventory).
+  // Then we look for a kubevirt-inventory__row-status-item which has a kubevirt-inventory__row-status-item-icon--ok
+  // under it, and go __back up__ to the parent (with "..") and find the span under it. This gives us
+  // the counter *next to* the kubevirt-inventory__row-status-item-icon--ok.
+  return element(by.xpath(`(//div[@class="kubevirt-inventory__row-status"])[${rowNumber + 1}]` +
+    '/div[contains(@class, "kubevirt-inventory__row-status-item")]' +
+    '/span[contains(@class, "kubevirt-inventory__row-status-item-icon--ok")]' +
+    '/../span[contains(@class, "kubevirt-inventory__row-status-item-text")]'));
+}
+
+/**
+ * The element in the inventory card, next to the down icon (red error icon) in the nth row.
+ * This element doesn't always exist, so check if it's present first, or use the getTextIfPresent
+ * utility function.
+ * @example
+ * await dashboardView.inventoryDownCounter(0).getText()
+ */
+export function inventoryDownCounter(rowNumber) {
+  // Similar xpath explanation to the one in inventoryUpCounter(), but now we're finding the counter
+  // next to the kubevirt-inventory__row-status-item-icon--error.
+  return element(by.xpath(`(//div[@class="kubevirt-inventory__row-status"])[${rowNumber + 1}]` +
+    '/div[contains(@class, "kubevirt-inventory__row-status-item")]' +
+    '/span[contains(@class, "kubevirt-inventory__row-status-item-icon--error")]' +
+    '/../span[contains(@class, "kubevirt-inventory__row-status-item-text")]'));
+}
+
+// substrings to identify the rows in the inventory card
+export const INVENTORY_NODES = ' Nodes';
+export const INVENTORY_HOSTS = ' Hosts';
+export const INVENTORY_PVCS = ' PVCs';
+export const INVENTORY_PODS = ' Pods';
+export const INVENTORY_VMS = ' VMs';
+export const INVENTORY_DISKS = ' Disks';
+
+/**
+ * Find the inventory row in the inventory card (depends on what you have installed)
+ * @example
+ * const rowNumber = await dashboardView.inventoryRow(dashboardView.INVENTORY_HOSTS);
+ */
+export const inventoryRow = async(substrInventory) => {
+  const elements = await $$('.kubevirt-inventory__row-title');
+  let i; // not using a forEach loop because you can't break or return from it
```

Comment:
I can't find such a function - can you link to some documentation of it?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex

Reply:
findIndex expects a filtering function which takes one parameter only (element). However, I want to filter the array by a certain substring - so I need the function to take 2 parameters (element, substrToMatch). It doesn't seem like findIndex lets me do that. Also, my filtering function would have to be async and I'm not sure findIndex will be able to call it with await.

Reply:
Using arrow functions solves the problem of passing the substring, but I really got stuck with the async issue because an async arrow function returns a Promise (and findIndex expects a function that returns a boolean).

Reply:
Ok we can use `for` if there are problems with `async` or if it looks ugly